### DISCUSSION TITLE
Dagster-dbt: using @dbt_assets with repository-based project

### DISCUSSION THREAD

**Discussion by YegorStolyarov** [dagster-io/dagster#21548] [Q&A] [answered]
While integrating dbt with an existing dagster project, I faced a problem with adding `DbtCliResource`.
`@d.repository` can't take resources, and also I would like to avoid migration from `@d.repository` to `d.Definitions`.

For some reason I don't know, `@dbt_assets` don't have `resource_defs` argument which could solve the problem. Under the hood, `@dbt_assets` could just pass the resource_defs argument to returning multi_asset and this could solve the problem.

Is there a way to add DbtCliResource resource manually to created dbt assets? I have tried `with_resources` function but without success.

**Comment by rexledesma** [comment, marked as answer]
See https://github.com/dagster-io/dagster/pull/21549#issuecomment-2088683166.
